### PR TITLE
docs: missing docs for tls termination for tcp

### LINKED ISSUE #1642: tls termination for tcp docs missing on latest
*Description*:
TLS termination was added in this [PR](https://github.com/envoyproxy/gateway/pull/1431) and docs was also added but it was not added to `user_docs.rst` file so it is not visible on the latest docs.

### PR DESCRIPTION
**Pull request by slayer321**
**What type of PR is this?**
docs: missing docs for tls termination for tcp

**What this PR does / why we need it**:
TLS termination was added in this [PR](https://github.com/envoyproxy/gateway/pull/1431) and docs was also added but it was not added to user_docs.rst file so it is not visible on the latest docs.
**Which issue(s) this PR fixes**:

Fixes #1642

### CODE REVIEW

File: docs/latest/user_docs.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ Learn how to deploy, use, and operate Envoy Gateway.
   user/secure-gateways
   user/tls-cert-manager
   user/tls-passthrough
```
Comment (arkodg) [inline docs/latest/user_docs.rst:19]: good catch, can you also update https://github.com/envoyproxy/gateway/blob/main/docs/v0.4.0/user_docs.rst

Comment (arkodg) [reply docs/latest/user_docs.rst:19]: if its relevant to this release (0.4.0), would be good to add it ^ as well

Comment (slayer321) [reply docs/latest/user_docs.rst:19]: As `0.4.0` version don't have that [file](https://github.com/envoyproxy/gateway/tree/main/docs/v0.4.0/user). So I think there is no need to add it to `0.4.0` version.

**Review by arkodg — APPROVED** [review]
LGTM, thanks !

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**